test/test_rpc.py

```diff
@@ -0,0 +1,181 @@
+
+class RpcTest(MultiProcessTestCase):
+
+    @property
+    def world_size(self):
+        return 4
+
+    @_wrap_with_rpc
+    def test_add(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, torch.add,
+                            args=(torch.ones(n, n), torch.ones(n, n)))
+        self.assertEqual(ret, torch.ones(n, n) * 2)
+
+    @_wrap_with_rpc
+    def test_scalar_add(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, torch.add,
+                            args=(torch.ones(n, n), n))
+        self.assertEqual(ret, (torch.ones(n, n) + n))
+
+    @_wrap_with_rpc
+    def test_async_add(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        fut = dist.rpc_async('worker%d' % dstRank, torch.add,
+                             args=(torch.ones(n, n), torch.ones(n, n)))
+        self.assertEqual(fut.wait(), torch.ones(n, n) * 2)
+
+    @_wrap_with_rpc
+    def test_nonzero(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        x = torch.ones(self.world_size, self.world_size)
+        x[self.rank][self.rank] = 0
+        ret = dist.rpc_sync('worker%d' % dstRank, torch.nonzero, args=(x,))
+        self.assertEqual(ret, x.nonzero())
+
+    @_wrap_with_rpc
+    def test_multi_rpc(self):
+        dstRank = (self.rank + 1) % self.world_size
+        for i in range(20):
+            n = i + self.rank + 1
+            ret = dist.rpc_sync('worker%d' % dstRank, torch.add,
+                                args=(torch.ones(n, n), torch.ones(n, n)))
+            self.assertEqual(ret, torch.ones(n, n) * 2)
+
+    @_wrap_with_rpc
+    def test_py_built_in(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, min, args=(n, n + 1, n + 2))
+        self.assertEqual(ret, n)
+
+    @_wrap_with_rpc
+    def test_py_user_defined(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, my_function, args=(n, n + 1, n + 2))
+        self.assertEqual(ret, (n + n + 1 + n + 2))
+
+    @_wrap_with_rpc
+    def test_py_class_constructor(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, my_class, args=(n,))
+        self.assertEqual(ret.a, n)
+
+    @_wrap_with_rpc
+    def test_py_class_instance_method(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, my_class(2).my_instance_method, args=(n,))
+        self.assertEqual(ret, (2 + n))
+
+    @_wrap_with_rpc
+    def test_py_class_method(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, my_class.my_class_method, args=(n, n + 1))
+        self.assertEqual(ret, (n + n + 1))
```

Comment:
Is there any reason for hard code the result instead of using the following?

```python
args = (n, n+1)
ret = dist.rpc_sync('worker%d' % dstRank, my_class.my_class_method, args=args)
self.assertEqual(ret, my_class.my_class_method(*args))
```

Reply:
I will change that